src/libraries/System.Net.Requests/src/System/Net/WebException.cs

```diff
@@ -96,18 +96,17 @@ internal static Exception CreateCompatibleException(Exception exception)
 
         private static WebExceptionStatus GetStatusFromExceptionHelper(HttpRequestException ex)
         {
-            SocketException? socketEx = ex.InnerException as SocketException;
+            NetworkException? networkException = ex.InnerException as NetworkException;
```

Comment:
Any concerns about the breaking change here for other consumers of HttpClient? i.e. if WebRequest is fishing out the inner exception and expecting it to be a SocketException and now it's a NetworkException?

Reply:
It's a minor concern, but I'm not sure there's a better option.

Note that this only matters for exceptions that come during Connect, since those are the only ones that have SocketException as the inner exception today. Other exceptions are already IOExceptions.

The ConnectionFactory stuff is generalizing the Connect path so that you can plug in your own transport. In general these can't throw SocketException, and that's what we invented NetworkException for. We could special case SocketConnectionFactory in SocketsHttpHandler, pull out the inner SocketException, and discard the NetworkException, but that means we'll have different exception behavior based on whether you use SocketConnectionFactory or not, which seems bad.

Long-term, a better solution here would be to have distinct HttpRequestException errors like HostNotFound so that a user can inspect it directly and doesn't need  to fish out the inner exception. WebRequestException already has this, as you can see here. We've discussed doing this in the past but haven't actually implemented it.